### PR TITLE
ESLint Parser Changes & Elaborated Types

### PR DESCRIPTION
**Pull request by andrewpeters9**
* I was running into some errors with `babel-eslint` - either it had TS support or its TS support was very limited
  * I migrated the parser to `@typescript-eslint`, which stopped my ESLint issues
* I made some types more explicit for the repo, esp. `typical.ts` to ensure that when I edit this file I don't make any unexpected breakages

### CODE REVIEW

File: src/typical.ts [hunk 1]
```diff
@@ -76,32 +82,40 @@ async function perform(
...
+
   while (startIndex < endIndex) {
-    yield text.slice(0, ++startIndex).join('');
+    yield splitText.slice(0, ++startIndex).join('');
   }
 }
 
-function* deleter([...text], startIndex = 0, endIndex = text.length) {
+function* deleter(text: string, startIndex = 0) {
+  const splitText = [...text];
```
Comment (andrewpeters9) [inline src/typical.ts:111]: New line created to facilitate splitter changes

File: package.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
     "build": "rollup -c",
     "start": "rollup -c -w",
     "prepare": "npm run build",
-    "react-start": "react-scripts start",
```
Comment (andrewpeters9) [inline package.json:19]: `npx react-scripts start` didn't seem to work for me. Let me know if you're still using it.

File: src/typical.ts [hunk 3]
```diff
@@ -76,32 +82,40 @@ async function perform(
...
-function* editor(edits: any) {
-  for (const edit of edits) {
+function* editor(edits: ReadonlyArray<string>) {
+  for (const snippet of edits) {
     yield {
-      op: (node: any) => requestAnimationFrame(() => (node.textContent = edit)),
+      op: (node: HTMLElementTagNameMap[Wrapper]) => requestAnimationFrame(() => (node.textContent = snippet)),
+
+      opCode: (node: HTMLElementTagNameMap[Wrapper]) => {
+        const nodeContent = node.textContent || '';
```
Comment (andrewpeters9) [inline src/typical.ts:91]: `node.textContent` can potentilly be `null` and therefore would error when `.length` is accessed on the line below.

**Comment by maxeth** [comment]
Hey, I appreciate all the work and improvements.

I noticed that you added the splitter functionality and removed it later on in https://github.com/maxeth/react-type-animation/pull/30/commits/befeae4c7c9bef6946486f18f6e4838f4b9779e7. Any background info on that? 😄 I tested the current state of your pr and the splitting bug with emojis is still present.

**Comment by andrewpeters9** [comment]
Hey @maxeth 

I removed those changes in order to make a separate PR for them - with this PR I was just intending to add some stricter typings, but let me know if you want them grouped together in the same one.

**Comment by maxeth** [comment]
> I removed those changes in order to make a separate PR for them

okay

**Pull request merged**